src/battle/area_trd_part_1/actor/bill_blaster.c

```diff
@@ -0,0 +1,250 @@
+    EVT_SWITCH(LW(0))
+        EVT_CASE_OR_EQ(EVENT_HIT_COMBO)
+        EVT_CASE_OR_EQ(EVENT_HIT)
+            EVT_SET_CONST(LW(0), 1)
+            EVT_SET_CONST(LW(1), NPC_ANIM_bill_blaster_Palette_00_Anim_3)
+            EVT_EXEC_WAIT(DoNormalHit)
+        EVT_END_CASE_GROUP
+        EVT_CASE_EQ(EVENT_BURN_HIT)
+            EVT_SET_CONST(LW(0), 1)
+            EVT_SET_CONST(LW(1), NPC_ANIM_bill_blaster_Palette_00_Anim_3)
+            EVT_SET_CONST(LW(2), -1)
+            EVT_EXEC_WAIT(DoBurnHit)
+        EVT_CASE_EQ(EVENT_BURN_DEATH)
+            EVT_SET_CONST(LW(0), 1)
+            EVT_SET_CONST(LW(1), NPC_ANIM_bill_blaster_Palette_00_Anim_3)
+            EVT_SET_CONST(LW(2), -1)
+            EVT_EXEC_WAIT(DoBurnHit)
+            EVT_SET_CONST(LW(0), 1)
+            EVT_SET_CONST(LW(1), NPC_ANIM_bill_blaster_Palette_00_Anim_3)
+            EVT_EXEC_WAIT(DoDeath)
+            EVT_RETURN
+        EVT_CASE_OR_EQ(EVENT_UNKNOWN_TRIGGER)
+        EVT_CASE_OR_EQ(EVENT_IMMUNE)
+        EVT_CASE_OR_EQ(EVENT_AIR_LIFT_FAILED)
+            EVT_SET_CONST(LW(0), 1)
+            EVT_SET_CONST(LW(1), NPC_ANIM_bill_blaster_Palette_00_Anim_1)
+            EVT_EXEC_WAIT(DoImmune)
+        EVT_END_CASE_GROUP
+        EVT_CASE_EQ(EVENT_DEATH)
+            EVT_SET_CONST(LW(0), 1)
+            EVT_SET_CONST(LW(1), NPC_ANIM_bill_blaster_Palette_00_Anim_3)
+            EVT_EXEC_WAIT(DoNormalHit)
+            EVT_WAIT_FRAMES(10)
+            EVT_SET_CONST(LW(0), 1)
+            EVT_SET_CONST(LW(1), NPC_ANIM_bill_blaster_Palette_00_Anim_3)
+            EVT_EXEC_WAIT(DoDeath)
+            EVT_RETURN
+        EVT_CASE_EQ(EVENT_RECOVER_STATUS)
+            EVT_SET_CONST(LW(0), 1)
+            EVT_SET_CONST(LW(1), NPC_ANIM_bill_blaster_Palette_00_Anim_1)
+            EVT_EXEC_WAIT(DoRecover)
+        EVT_CASE_EQ(EVENT_BEGIN_AIR_LIFT)
+            EVT_SET_CONST(LW(0), 1)
+            EVT_SET_CONST(LW(1), NPC_ANIM_bill_blaster_Palette_00_Anim_3)
+            EVT_EXEC_WAIT(DoAirLift)
+        EVT_CASE_EQ(EVENT_BLOW_AWAY)
+            EVT_SET_CONST(LW(0), 1)
+            EVT_SET_CONST(LW(1), NPC_ANIM_bill_blaster_Palette_00_Anim_3)
+            EVT_EXEC_WAIT(DoBlowAway)
+            EVT_RETURN
+        EVT_CASE_DEFAULT
+    EVT_END_SWITCH
+    EVT_CALL(EnableIdleScript, ACTOR_SELF, 1)
+    EVT_CALL(UseIdleAnimation, ACTOR_SELF, TRUE)
+    EVT_RETURN
+    EVT_END
+};
+
+extern EvtSource N(8021A2BC);
+extern EvtSource N(8021A470);
+
+EvtSource N(takeTurn_8021A200) = {
+    EVT_CALL(UseIdleAnimation, ACTOR_SELF, FALSE)
+    EVT_CALL(EnableIdleScript, ACTOR_SELF, 0)
+    EVT_CALL(GetActorVar, ACTOR_SELF, 8, LW(0))
+    EVT_SWITCH(LW(0))
+        EVT_CASE_EQ(0)
+            EVT_EXEC_WAIT(N(8021A2BC))
+        EVT_CASE_EQ(1)
+            EVT_EXEC_WAIT(N(8021A470))
+    EVT_END_SWITCH
+    EVT_CALL(EnableIdleScript, ACTOR_SELF, 1)
+    EVT_CALL(UseIdleAnimation, ACTOR_SELF, TRUE)
+    EVT_RETURN
+    EVT_END
+};
+
+extern Formation N(specialFormation_8021A4E8);
```

Comment:
I think these externs should all be moved to the top

Reply:
I can do that, but be aware that I haven't done this for almost any of these files. I've been putting them above their usages

Reply:
It'll probably be worth moving them all in one go later, feel free to resolve without changes if you want

Reply:
I think we should do this sooner rather than later. I can try to go through the older PRs and do it soon, but going forward we probably should